{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Projeto",
      "provenance": [],
      "collapsed_sections": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Sammypacoo/Rosalind/blob/main/Projeto.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 542
        },
        "id": "xJ3DEAIvT4GI",
        "outputId": "498964f1-7140-4f6e-aa69-6e95a2334df0"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<html>\n",
              "<head><meta charset=\"utf-8\" /></head>\n",
              "<body>\n",
              "    <div>            <script src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG\"></script><script type=\"text/javascript\">if (window.MathJax) {MathJax.Hub.Config({SVG: {font: \"STIX-Web\"}});}</script>                <script type=\"text/javascript\">window.PlotlyConfig = {MathJaxConfig: 'local'};</script>\n",
              "        <script src=\"https://cdn.plot.ly/plotly-2.8.3.min.js\"></script>                <div id=\"a6698b22-0a0e-4467-8711-7fc983ec5df5\" class=\"plotly-graph-div\" style=\"height:525px; width:100%;\"></div>            <script type=\"text/javascript\">                                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById(\"a6698b22-0a0e-4467-8711-7fc983ec5df5\")) {                    Plotly.newPlot(                        \"a6698b22-0a0e-4467-8711-7fc983ec5df5\",                        [{\"fill\":\"toself\",\"name\":\"APP\",\"r\":[12.9,9.09],\"theta\":[\"GNOMAD\",\"SABE\"],\"type\":\"scatterpolar\"},{\"fill\":\"toself\",\"name\":\"MAPT\",\"r\":[27.8,19.3],\"theta\":[\"GNOMAD\",\"SABE\"],\"type\":\"scatterpolar\"},{\"fill\":\"toself\",\"name\":\"PSEN1\",\"r\":[13,0],\"theta\":[\"GNOMAD\",\"SABE\"],\"type\":\"scatterpolar\"},{\"fill\":\"toself\",\"name\":\"PSEN2\",\"r\":[10.4,0],\"theta\":[\"GNOMAD\",\"SABE\"],\"type\":\"scatterpolar\"},{\"fill\":\"toself\",\"name\":\"TREM2\",\"r\":[6.2,5],\"theta\":[\"GNOMAD\",\"SABE\"],\"type\":\"scatterpolar\"}],                        {\"template\":{\"data\":{\"bar\":[{\"error_x\":{\"color\":\"#2a3f5f\"},\"error_y\":{\"color\":\"#2a3f5f\"},\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"bar\"}],\"barpolar\":[{\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"barpolar\"}],\"carpet\":[{\"aaxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"baxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"type\":\"carpet\"}],\"choropleth\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"type\":\"choropleth\"}],\"contour\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"contour\"}],\"contourcarpet\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"type\":\"contourcarpet\"}],\"heatmap\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"heatmap\"}],\"heatmapgl\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"heatmapgl\"}],\"histogram\":[{\"marker\":{\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"histogram\"}],\"histogram2d\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"histogram2d\"}],\"histogram2dcontour\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"histogram2dcontour\"}],\"mesh3d\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"type\":\"mesh3d\"}],\"parcoords\":[{\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"parcoords\"}],\"pie\":[{\"automargin\":true,\"type\":\"pie\"}],\"scatter\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scatter\"}],\"scatter3d\":[{\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scatter3d\"}],\"scattercarpet\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scattercarpet\"}],\"scattergeo\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scattergeo\"}],\"scattergl\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scattergl\"}],\"scattermapbox\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scattermapbox\"}],\"scatterpolar\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scatterpolar\"}],\"scatterpolargl\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scatterpolargl\"}],\"scatterternary\":[{\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"type\":\"scatterternary\"}],\"surface\":[{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"type\":\"surface\"}],\"table\":[{\"cells\":{\"fill\":{\"color\":\"#EBF0F8\"},\"line\":{\"color\":\"white\"}},\"header\":{\"fill\":{\"color\":\"#C8D4E3\"},\"line\":{\"color\":\"white\"}},\"type\":\"table\"}]},\"layout\":{\"annotationdefaults\":{\"arrowcolor\":\"#2a3f5f\",\"arrowhead\":0,\"arrowwidth\":1},\"autotypenumbers\":\"strict\",\"coloraxis\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"colorscale\":{\"diverging\":[[0,\"#8e0152\"],[0.1,\"#c51b7d\"],[0.2,\"#de77ae\"],[0.3,\"#f1b6da\"],[0.4,\"#fde0ef\"],[0.5,\"#f7f7f7\"],[0.6,\"#e6f5d0\"],[0.7,\"#b8e186\"],[0.8,\"#7fbc41\"],[0.9,\"#4d9221\"],[1,\"#276419\"]],\"sequential\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"sequentialminus\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]},\"colorway\":[\"#636efa\",\"#EF553B\",\"#00cc96\",\"#ab63fa\",\"#FFA15A\",\"#19d3f3\",\"#FF6692\",\"#B6E880\",\"#FF97FF\",\"#FECB52\"],\"font\":{\"color\":\"#2a3f5f\"},\"geo\":{\"bgcolor\":\"white\",\"lakecolor\":\"white\",\"landcolor\":\"#E5ECF6\",\"showlakes\":true,\"showland\":true,\"subunitcolor\":\"white\"},\"hoverlabel\":{\"align\":\"left\"},\"hovermode\":\"closest\",\"mapbox\":{\"style\":\"light\"},\"paper_bgcolor\":\"white\",\"plot_bgcolor\":\"#E5ECF6\",\"polar\":{\"angularaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"bgcolor\":\"#E5ECF6\",\"radialaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"scene\":{\"xaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"gridwidth\":2,\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\"},\"yaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"gridwidth\":2,\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\"},\"zaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"gridwidth\":2,\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\"}},\"shapedefaults\":{\"line\":{\"color\":\"#2a3f5f\"}},\"ternary\":{\"aaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"baxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"bgcolor\":\"#E5ECF6\",\"caxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"title\":{\"x\":0.05},\"xaxis\":{\"automargin\":true,\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"zerolinewidth\":2},\"yaxis\":{\"automargin\":true,\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"zerolinewidth\":2}}},\"polar\":{\"radialaxis\":{\"visible\":true,\"range\":[0,50]}},\"showlegend\":true},                        {\"responsive\": true}                    ).then(function(){\n",
              "                            \n",
              "var gd = document.getElementById('a6698b22-0a0e-4467-8711-7fc983ec5df5');\n",
              "var x = new MutationObserver(function (mutations, observer) {{\n",
              "        var display = window.getComputedStyle(gd).display;\n",
              "        if (!display || display === 'none') {{\n",
              "            console.log([gd, 'removed!']);\n",
              "            Plotly.purge(gd);\n",
              "            observer.disconnect();\n",
              "        }}\n",
              "}});\n",
              "\n",
              "// Listen for the removal of the full notebook cells\n",
              "var notebookContainer = gd.closest('#notebook-container');\n",
              "if (notebookContainer) {{\n",
              "    x.observe(notebookContainer, {childList: true});\n",
              "}}\n",
              "\n",
              "// Listen for the clearing of the current output cell\n",
              "var outputEl = gd.closest('.output');\n",
              "if (outputEl) {{\n",
              "    x.observe(outputEl, {childList: true});\n",
              "}}\n",
              "\n",
              "                        })                };                            </script>        </div>\n",
              "</body>\n",
              "</html>"
            ]
          },
          "metadata": {}
        }
      ],
      "source": [
        "import plotly.graph_objects as go\n",
        "\n",
        "categories = ['GNOMAD', 'SABE']\n",
        "\n",
        "fig = go.Figure()\n",
        "\n",
        "\n",
        "fig.add_trace(go.Scatterpolar(\n",
        "      r=[12.9, 9.09],\n",
        "      theta=categories,\n",
        "      fill='toself',\n",
        "      name='APP'\n",
        "))\n",
        "fig.add_trace(go.Scatterpolar(\n",
        "      r=[27.8, 19.3],\n",
        "      theta=categories,\n",
        "      fill='toself',\n",
        "      name='MAPT'\n",
        "))\n",
        "\n",
        "fig.add_trace(go.Scatterpolar(\n",
        "      r=[13,0],\n",
        "      theta=categories,\n",
        "      fill='toself',\n",
        "      name='PSEN1'\n",
        "))\n",
        "fig.add_trace(go.Scatterpolar(\n",
        "      r=[10.4,0],\n",
        "      theta=categories,\n",
        "      fill='toself',\n",
        "      name='PSEN2'\n",
        "))\n",
        "fig.add_trace(go.Scatterpolar(\n",
        "      r=[ 6.2,5],\n",
        "      theta=categories,\n",
        "      fill='toself',\n",
        "      name='TREM2'\n",
        "))\n",
        "\n",
        "\n",
        "fig.update_layout(\n",
        "  polar=dict(\n",
        "    radialaxis=dict(\n",
        "      visible=True,\n",
        "      range=[0, 50]\n",
        "    )),\n",
        "  showlegend=True\n",
        ")\n",
        "\n",
        "fig.show()"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DDwJiH0oVqYp",
        "outputId": "0b2d2484-a60f-4066-b097-cfd0b51144d7"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from collections import defaultdict\n",
        "import gzip\n",
        "import pandas as pd\n",
        "import re"
      ],
      "metadata": {
        "id": "_gHQ0f2EE6iR"
      },
      "execution_count": 6,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Funções do código\n"
      ],
      "metadata": {
        "id": "T9GmzLPxSFwU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def Convert(tup, di):\n",
        "    for a, b in tup:\n",
        "        di.setdefault(a, []).append(b)\n",
        "    return di\n",
        "      \n",
        "# Driver Code    \n",
        "tups = [(\"akash\", 10), (\"gaurav\", 12), (\"anand\", 14), \n",
        "     (\"suraj\", 20), (\"akhil\", 25), (\"ashish\", 30)]\n",
        "dictionary = {}\n",
        "print (Convert(tups, dictionary))"
      ],
      "metadata": {
        "id": "IttgFaF4OTKe",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "93e2630f-c370-442c-f3ea-2396e6b312c1"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "{'akash': [10], 'gaurav': [12], 'anand': [14], 'suraj': [20], 'akhil': [25], 'ashish': [30]}\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "variantes=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_do_projeto/app_gnomad.csv\", sep=';')\n",
        "variantes.columns\n",
        "\n",
        "\n",
        "\n",
        "arq1=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_importantes_p_codigo/CCDS.current.txt\" , sep='\\t')\n",
        "\n",
        "\n",
        "arq2=arquivo=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_importantes_p_codigo/CCDS2Sequence.current.txt\" , sep='\\t')\n",
        "\n",
        "\n",
        "arq3=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_importantes_p_codigo/exon_seq1\", header= None , sep='\\t')\n",
        "# criei uma tabela com os dados CCDS_protein_exons.current.faa\"\n",
        "\n",
        "df=pd.read_table(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_importantes_p_codigo/GRCh38_latest_genomic.gtf\",header=None,skiprows=4)\n",
        "\n"
      ],
      "metadata": {
        "id": "yBp6NSQ67ZX4"
      },
      "execution_count": 8,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "\n",
        "def tab_final(gene):\n",
        "\n",
        "    arq1.shape\n",
        "    arq1.rename(columns={\"ccds_id\":\"#ccds\"}, inplace = True)\n",
        "    arq1.columns\n",
        "    app=arq1[arq1[\"gene\"]==gene]\n",
        "    app.shape\n",
        "    app.columns\n",
        "    app\n",
        "\n",
        "    result = pd.merge(app, arq2, on=\"#ccds\",how=\"left\")\n",
        "    result = result.drop_duplicates()\n",
        "    result=result[result[\"source\"]==\"NCBI\"]\n",
        "    result=result[result[\"status_in_CCDS\"]==\"Accepted\"]\n",
        "    result\n",
        "    return result     "
      ],
      "metadata": {
        "id": "GwEh87NpPpBe"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Para cada posição do genomica, mostra tods os NM relacionados aquela posi genomica\n",
        "def tab_teste1(final,x):\n",
        "  x=int(x)\n",
        "\n",
        "  # Fazer listas das extremidades do gene\n",
        "\n",
        "  ext1=final.iloc[:,7].tolist()\n",
        "  ext2=final.iloc[:,8].tolist()\n",
        "  lista_fil1=[]\n",
        "  for j in range (len(ext1)):\n",
        "      x1=ext1[j]\n",
        "      x1=int(x1)+1\n",
        "      x2=ext2[j]\n",
        "      x2=int(x2)+2\n",
        "      r=range(x1,x2)\n",
        "\n",
        "      if x in r:\n",
        "          lista_fil1+=[1]\n",
        "      else:\n",
        "          lista_fil1+=[0]\n",
        "  teste=final\n",
        "  teste[\"filtro1\"]=lista_fil1\n",
        "  teste=teste[teste[\"filtro1\"]==1]\n",
        "\n",
        "\n",
        "  # Verificar em  quais nms aquela posi genômica está presente \n",
        "  tam1=teste.shape\n",
        "  tam1=tam1[0]\n",
        "  lista_fil2=[]\n",
        "  for xl in range(tam1):\n",
        "      f1=teste.iloc[xl,9]\n",
        "      f1=str(f1)\n",
        "      f1=re.findall(r'[0-9]+', f1)\n",
        "      for xl1 in range (0,len(f1)-1,2):\n",
        "          ini=f1[xl1]\n",
        "          ini=int(ini)+1\n",
        "          final=f1[xl1+1]\n",
        "          final=int(final)+2\n",
        "          r3=range(ini,final)\n",
        "          if x in r3:\n",
        "            lista_fil2+=[1] \n",
        "            break\n",
        "          else:\n",
        "            if xl1 == len(f1)-2:\n",
        "                lista_fil2+=[0]\n",
        "           \n",
        "\n",
        "  teste[\"filtro2\"]=lista_fil2\n",
        "  teste1=teste[teste[\"filtro2\"]==1]\n",
        "\n",
        "     \n",
        "  #teste1.rename(columns={5:\"#ccds\"}, inplace = True)\n",
        "  return teste1"
      ],
      "metadata": {
        "id": "budOn0OKQPIJ"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# final seria o teste1\n",
        "def tab_exons_seq(final):\n",
        "  taman=final.shape\n",
        "  taman=taman[0]\n",
        "  if taman >0:\n",
        "    df=pd.DataFrame()  \n",
        "    dic={}\n",
        "    dic2={}\n",
        "    g2=[]\n",
        "    for i in range (taman):\n",
        "      strand=final.iloc[i,6]\n",
        "      dic={}\n",
        "      bb=final.iloc[i,9]\n",
        "      tf=bb.replace(\"[\",\"\")\n",
        "      tf=tf.replace(\"]\",\"\")\n",
        "      tf=tf.split(\",\")\n",
        "      for i2 in range(len(tf)):    \n",
        "        p=tf[i2]\n",
        "        p=p.split(\"-\")\n",
        "        p=p[0]\n",
        "        p=int(p)\n",
        "        dic[tf[i2]]=p\n",
        "      d_descending =sorted(dic.items(), key=lambda x: x[1], reverse=True)\n",
        "      g1=dic.keys()\n",
        "      g1=list(g1)\n",
        "      dic2={}\n",
        "      Convert(d_descending, dic2)\n",
        "      g2=dic2.keys()\n",
        "      g2=list(g2)\n",
        "      g=final.iloc[i,4]\n",
        "      if strand==\"-\":\n",
        "        strand='-'\n",
        "        y=arq3[arq3[0]==g]\n",
        "        y[\"éxon\"]=g2\n",
        "      else:\n",
        "        strand='+'\n",
        "        y=arq3[arq3[0]==g]\n",
        "        y[\"éxon\"]=g1\n",
        "\n",
        "      \n",
        "      lin=y.shape\n",
        "      lin=lin[0]\n",
        "      list_strand=[]\n",
        "      for it in range(lin):\n",
        "          list_strand+=[strand]\n",
        "      y[\"strand\"]=list_strand\n",
        "      df=pd.concat([df,y],axis=0)\n",
        "  else:\n",
        "     df=pd.DataFrame() \n",
        "\n",
        "  return df"
      ],
      "metadata": {
        "id": "F_hO2MHVPRiC"
      },
      "execution_count": 11,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "def resul(teste1,df,x):\n",
        "  x=int(x)\n",
        "  taman=df.shape\n",
        "  taman=taman[0]\n",
        "  if taman >0:\n",
        "    lis=df.iloc[:,0].unique()\n",
        "    lis=lis.tolist()\n",
        "\n",
        "    l1=[]\n",
        "    #colocar o for\n",
        "    for jkk in range(len(lis)):\n",
        "      f=df[df[0]==lis[jkk]]\n",
        "      ggp=f[4].tolist()\n",
        "      listp=[]\n",
        "      for j in range (len(ggp)):\n",
        "        ggp1=len(ggp[j])\n",
        "        listp+=[ggp1]\n",
        "      f[\"tamanho\"]=listp\n",
        "\n",
        "      listp2=[0]\n",
        "      valor=0\n",
        "      for j1 in range (len(listp)-1):\n",
        "        valor=valor + listp[j1]\n",
        "        listp2+=[valor]\n",
        "      f[\"Start_Value\"]=listp2\n",
        "      l=[]    \n",
        "      for j2 in range (len(listp)):\n",
        "        ran=f.iloc[j2,5]\n",
        "        ran=ran.split(\"-\")\n",
        "        a1=int(ran[0])+1\n",
        "        b1=int(ran[1])+2\n",
        "        r=range(a1,b1)\n",
        "        if x in r:\n",
        "          l+=[1]\n",
        "        else:\n",
        "          l+=[0]\n",
        "\n",
        "      f[\"filtro\"]=l\n",
        "      gg3=f[f[\"filtro\"]==1]\n",
        "\n",
        "      strand=gg3.iloc[0,6]\n",
        "      ran=gg3.iloc[0,5]\n",
        "\n",
        "      cc=ran.split(\"-\")\n",
        "      a1=int(cc[0])+1\n",
        "      b1=int(cc[1])+1\n",
        "      \n",
        "\n",
        "      if strand==\"-\":\n",
        "        d=b1-x       \n",
        "        s=(d//3)+1\n",
        "      else:\n",
        "        a1=a1-1\n",
        "        d=x-a1\n",
        "        \n",
        "        s=(d//3)\n",
        "        \n",
        "\n",
        "        \n",
        "\n",
        "      s1=gg3.iloc[0,8]\n",
        "      \n",
        "      s,s1\n",
        "      res=s+s1\n",
        "      res\n",
        "      l1+=[res]\n",
        "\n",
        "\n",
        "    nm=teste1.iloc[:,14]\n",
        "    nm=nm.tolist()\n",
        "\n",
        "    gg4=pd.DataFrame(nm,columns=[\"Nm\"])\n",
        "    gg5=pd.DataFrame(l1,columns=[\"Pos\"])\n",
        "    gg6=pd.concat([gg4,gg5],axis=1)\n",
        "    tamk=gg6.shape\n",
        "    tamk=tamk[0]\n",
        "    listaf=[]\n",
        "    for ih in range (tamk):\n",
        "      it=gg6.iloc[ih,:]\n",
        "      it=it.tolist()\n",
        "      listaf+=[it]\n",
        "  else:\n",
        "    listaf=[]\n",
        "  return listaf"
      ],
      "metadata": {
        "id": "lo8lDXpDOHtq"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import re \n",
        "\n",
        "def sitios(X,gene):\n",
        "    if len(X)>0:\n",
        "      ptm=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_do_projeto/Arquivos_importantes/phosphositeplus_allptms_REFSEQ1\", sep='\\t')\n",
        "      ptm.columns\n",
        "\n",
        "      ptm=ptm[ptm[\"GENE\"]==gene]\n",
        "      ptm.dropna(subset = [\"Refseq_semdigito\"], inplace=True)\n",
        "      nm_sitio=ptm[\"Refseq_semdigito\"].tolist()\n",
        "      tes=ptm[\"Refseq_semdigito\"].unique()\n",
        "      lista=[]\n",
        "      for cd in range (len(nm_sitio)):\n",
        "          x=nm_sitio[cd]\n",
        "          x=str(x)\n",
        "          x1=re.findall(r'\\bNM_\\w+',x)\n",
        "          x1=x1[0]\n",
        "          lista+=[x1]\n",
        "      \n",
        "      \n",
        "      ptm[\"R\"]=lista\n",
        "      pos_sitio=ptm[\"SITE\"].tolist()\n",
        "      lista2=[]\n",
        "      for cd1 in range (len(pos_sitio)):\n",
        "          #print(cd1)\n",
        "          x=pos_sitio[cd1]\n",
        "          x=str(x)\n",
        "          x1=re.findall(r'[0-9]+',x)\n",
        "          x1=x1[0]\n",
        "          x1=int(x1)\n",
        "          #print(x1)\n",
        "          lista2+=[x1]\n",
        "      len(lista2)\n",
        "      ptm[\"Posição\"]=lista2\n",
        "      s=\"n\"\n",
        "      lista4=[]\n",
        "      for i in range (len(X)):\n",
        "          nm=X[i][0]\n",
        "          nm=re.findall(r'\\bNM_\\w+',nm)\n",
        "          nm=nm[0]\n",
        "          pos=X[i][1]\n",
        "          pos=str(pos)\n",
        "          pos=re.findall(r'[0-9]+', pos)\n",
        "          pos=pos[0]\n",
        "          pos=int(pos)\n",
        "          \n",
        "          A=pos+1\n",
        "          a=pos-1\n",
        "          lista1=[a,pos,A]\n",
        "          t=-2  \n",
        "          nm_sitio=ptm[ptm[\"R\"]==nm]\n",
        "      \n",
        "          tam=nm_sitio.shape\n",
        "          tam=tam[0]\n",
        "          if tam!=0:\n",
        "              #print(\"entrei\")\n",
        "                    \n",
        "              for v in range (len(lista1)):\n",
        "                  sitio_p=lista1[v]\n",
        "                  #print(sitio_p)\n",
        "                  t+=1\n",
        "      \n",
        "                  p_sitio=nm_sitio[nm_sitio[\"Posição\"]==sitio_p]\n",
        "                  #print(p_sitio)\n",
        "      \n",
        "                  tam1=p_sitio.shape\n",
        "                  tam1=tam1[0]\n",
        "                  if tam1!=0:\n",
        "                      #print(\"entrei1\")                \n",
        "                      s=\"s\"\n",
        "                      tam3=p_sitio.shape\n",
        "                      tam3=tam3[0]\n",
        "                      lista3=[]\n",
        "                      for k in range (tam3):\n",
        "                          row=p_sitio.iloc[:,[0,1,12,13]]\n",
        "                          row=row.iloc[k,:].tolist()\n",
        "                          row+=[t]\n",
        "                          \n",
        "                          lista3+=[row]\n",
        "                      lista4+=[lista3]\n",
        "                    \n",
        "                      break\n",
        "\n",
        "    else : \n",
        "      s=\"n\"\n",
        "      lista4=[]                     \n",
        "    return s, lista4"
      ],
      "metadata": {
        "id": "vtASB1S9B3lp"
      },
      "execution_count": 13,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#total"
      ],
      "metadata": {
        "id": "-xT15U7zNL87"
      }
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "UElGWKxYY_wy"
      },
      "execution_count": 13,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "dic={\"APP\":\"/content/drive/MyDrive/Projeto/Projeto/Arquivos_do_projeto/app_gnomad.csv\",\n",
        "     \"MAPT\":\"/content/drive/MyDrive/Projeto/Projeto/MAPT_gnomad.csv\",\n",
        "     \"PSEN1\":\"/content/drive/MyDrive/Projeto/Projeto/PSEN1_gnomad.csv\",\n",
        "     \"PSEN2\":\"/content/drive/MyDrive/Projeto/Projeto/PSEN2_gnomad.csv\",\n",
        "     \"TREM2\":\"/content/drive/MyDrive/Projeto/Projeto/TREM2_gnomad.csv\"}\n",
        "dic_re={}\n",
        "df_resul_gnomad=pd.DataFrame(columns=[\"Gene\",\"Gnomad Total\",\"Gnomad_ptm\",\"Gnomad_Porc\"])\n",
        "\n",
        "\n",
        "for chave in dic :\n",
        "  if chave==\"APP\":\n",
        "    variantes=pd.read_csv(dic[chave], sep=';')\n",
        "  else:\n",
        "    variantes=pd.read_csv(dic[chave], sep=',')\n",
        "  gene=chave\n",
        "  final=tab_final(gene)\n",
        "  final\n",
        "  variantes.head()\n",
        "  variantes=variantes[variantes[\"VEP Annotation\"]==\"missense_variant\"]\n",
        "  variantes\n",
        "  list_x=variantes[\"Position\"].tolist()\n",
        "\n",
        "  tk=[]\n",
        "  tk1=[]\n",
        "  tk2=[]\n",
        "  for hgh in range(len(list_x)):\n",
        "    print(hgh)  \n",
        "    x=list_x[hgh]\n",
        "\n",
        "    teste1=tab_teste1(final,x)\n",
        "    teste1\n",
        "\n",
        "    df=tab_exons_seq(teste1)\n",
        "    #print(df)\n",
        "    lista=resul(teste1,df,x)\n",
        "\n",
        "    a,b=sitios(lista,gene)\n",
        "    tk+=[a]\n",
        "    tk1+=[b]\n",
        "    tk2+=[lista]\n",
        "  print(tk)\n",
        "  variantes[\"PTM\"]=tk\n",
        "  variantes[\"PTM_INF\"]=tk1\n",
        "  variantes[\"Variantes\"]=tk2\n",
        "  variantes_gnomad_sabe=variantes\n",
        "  variantes_sito=variantes[variantes[\"PTM\"]==\"s\"]\n",
        "  dic_re[chave]=variantes_sito\n",
        "  X=len(variantes)\n",
        "  X1=len(variantes_sito)\n",
        "  X2=(X1*100)/X\n",
        "  X,X1,X2\n",
        "  lg=[chave,X,X1,X2]\n",
        "  df_resul_gnomad.loc[len(df_resul_gnomad)] = lg\n",
        "  df_resul_gnomad\n"
      ],
      "metadata": {
        "id": "YHiBA2vlM9qT"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "  df_resul_gnomad"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "YTmt91ziZCVK",
        "outputId": "6d3528d4-02be-44eb-b677-6a482a3ac28b"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Gene Gnomad Total Gnomad_ptm  Gnomad_Porc\n",
              "0    APP          263         34    12.927757\n",
              "1   MAPT          355         99    27.887324\n",
              "2  PSEN1          138         18    13.043478\n",
              "3  PSEN2          191         20    10.471204\n",
              "4  TREM2          120          6     5.000000"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-025fdc52-36f2-4034-a76e-e90de6c5be98\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Gnomad Total</th>\n",
              "      <th>Gnomad_ptm</th>\n",
              "      <th>Gnomad_Porc</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>APP</td>\n",
              "      <td>263</td>\n",
              "      <td>34</td>\n",
              "      <td>12.927757</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>MAPT</td>\n",
              "      <td>355</td>\n",
              "      <td>99</td>\n",
              "      <td>27.887324</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PSEN1</td>\n",
              "      <td>138</td>\n",
              "      <td>18</td>\n",
              "      <td>13.043478</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PSEN2</td>\n",
              "      <td>191</td>\n",
              "      <td>20</td>\n",
              "      <td>10.471204</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>TREM2</td>\n",
              "      <td>120</td>\n",
              "      <td>6</td>\n",
              "      <td>5.000000</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-025fdc52-36f2-4034-a76e-e90de6c5be98')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-025fdc52-36f2-4034-a76e-e90de6c5be98 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-025fdc52-36f2-4034-a76e-e90de6c5be98');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "dic_re"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1_46pqViIToL",
        "outputId": "f5b2d23f-4907-499f-83d2-f777c45f8fe8"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'APP':      Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
              " 0            21  25881702   rs766455623         T         C  gnomAD Genomes   \n",
              " 1            21  25881705  rs1300518717         G         T  gnomAD Genomes   \n",
              " 3            21  25881734  rs1437503550         G         C  gnomAD Genomes   \n",
              " 29           21  25897632  rs1259157720         C         T  gnomAD Genomes   \n",
              " 30           21  25897641   rs200084346         T         A  gnomAD Genomes   \n",
              " 31           21  25897642    rs63750363         C         G  gnomAD Genomes   \n",
              " 33           21  25897649   rs200260102         G         A  gnomAD Genomes   \n",
              " 34           21  25897655   rs201104127         A         G  gnomAD Genomes   \n",
              " 35           21  25897659   rs770773038         T         C  gnomAD Genomes   \n",
              " 36           21  25897667           NaN         C         G  gnomAD Genomes   \n",
              " 37           21  25897668  rs1305531440         C         T  gnomAD Genomes   \n",
              " 54           21  25911849   rs201724379         T         C  gnomAD Genomes   \n",
              " 55           21  25911851   rs200088099         G         A  gnomAD Genomes   \n",
              " 56           21  25911855   rs140304729         C         T  gnomAD Genomes   \n",
              " 85           21  25954656           NaN         T         C  gnomAD Genomes   \n",
              " 102          21  25955728   rs201384815         T         G  gnomAD Genomes   \n",
              " 116          21  25975122   rs199549173         C         T  gnomAD Genomes   \n",
              " 117          21  25975123   rs200548072         G         A  gnomAD Genomes   \n",
              " 118          21  25975125   rs746164396         C         T  gnomAD Genomes   \n",
              " 119          21  25975126  rs1162419578         G         A  gnomAD Genomes   \n",
              " 120          21  25975129  rs1424012655         C         A  gnomAD Genomes   \n",
              " 124          21  25975203   rs749605439         A         G  gnomAD Genomes   \n",
              " 127          21  25975966  rs1418014426         C         A  gnomAD Genomes   \n",
              " 128          21  25975967   rs188433801         T         C  gnomAD Genomes   \n",
              " 138          21  25982424   rs752243493         G         A  gnomAD Genomes   \n",
              " 139          21  25982427  rs1322571887         T         C  gnomAD Genomes   \n",
              " 140          21  25982430   rs755703063         C         T  gnomAD Genomes   \n",
              " 142          21  25982462  rs1193124736         G         T  gnomAD Genomes   \n",
              " 143          21  25982472   rs748168213         T         C  gnomAD Genomes   \n",
              " 144          21  25997360   rs749453173         G         C  gnomAD Genomes   \n",
              " 145          21  25997360   rs749453173         G         A  gnomAD Genomes   \n",
              " 215          21  26051045   rs376138420         G         A  gnomAD Genomes   \n",
              " 221          21  26051070   rs145081708         A         G  gnomAD Genomes   \n",
              " 228          21  26051129   rs758637653         T         C  gnomAD Genomes   \n",
              " \n",
              "      Filters - exomes Filters - genomes         Transcript HGVS Consequence  \\\n",
              " 0                 NaN              PASS  ENST00000346798.8      p.Thr761Ala   \n",
              " 1                 NaN              PASS  ENST00000346798.8      p.Pro760Thr   \n",
              " 3                 NaN              PASS  ENST00000346798.8      p.Ser750Cys   \n",
              " 29                NaN              PASS  ENST00000346798.8      p.Val669Met   \n",
              " 30                NaN              PASS  ENST00000346798.8      p.Ile666Phe   \n",
              " 31                NaN              PASS  ENST00000346798.8      p.Glu665Asp   \n",
              " 33                NaN              PASS  ENST00000346798.8      p.Thr663Met   \n",
              " 34                NaN              PASS  ENST00000346798.8      p.Ile661Thr   \n",
              " 35                NaN              PASS  ENST00000346798.8      p.Asn660Asp   \n",
              " 36                NaN              PASS  ENST00000346798.8      p.Gly657Ala   \n",
              " 37                NaN              PASS  ENST00000346798.8      p.Gly657Arg   \n",
              " 54                NaN              PASS  ENST00000346798.8      p.Lys601Glu   \n",
              " 55                NaN              PASS  ENST00000346798.8      p.Thr600Met   \n",
              " 56                NaN              PASS  ENST00000346798.8      p.Glu599Lys   \n",
              " 85                NaN              PASS  ENST00000346798.8      p.Met541Val   \n",
              " 102               NaN              PASS  ENST00000346798.8      p.Lys496Gln   \n",
              " 116               NaN              PASS  ENST00000346798.8      p.Arg469His   \n",
              " 117               NaN              PASS  ENST00000346798.8      p.Arg469Cys   \n",
              " 118               NaN              PASS  ENST00000346798.8      p.Arg468His   \n",
              " 119               NaN              PASS  ENST00000346798.8      p.Arg468Cys   \n",
              " 120               NaN              PASS  ENST00000346798.8      p.Asp467Tyr   \n",
              " 124               NaN              PASS  ENST00000346798.8      p.Leu442Ser   \n",
              " 127               NaN              PASS  ENST00000346798.8      p.Lys429Asn   \n",
              " 128               NaN              PASS  ENST00000346798.8      p.Lys429Arg   \n",
              " 138               NaN              PASS  ENST00000346798.8      p.Pro382Ser   \n",
              " 139               NaN              PASS  ENST00000346798.8      p.Thr381Ala   \n",
              " 140               NaN              PASS  ENST00000346798.8      p.Glu380Lys   \n",
              " 142               NaN              PASS  ENST00000346798.8      p.Ala369Asp   \n",
              " 143               NaN              PASS  ENST00000346798.8      p.Thr366Ala   \n",
              " 144               NaN              PASS  ENST00000346798.8      p.Leu364Val   \n",
              " 145               NaN              PASS  ENST00000346798.8      p.Leu364Phe   \n",
              " 215               NaN              PASS  ENST00000346798.8      p.Ser206Leu   \n",
              " 221               NaN              PASS  ENST00000346798.8      p.Ser198Pro   \n",
              " 228               NaN              PASS  ENST00000346798.8      p.Lys178Arg   \n",
              " \n",
              "      ... Allele Number Ashkenazi Jewish Homozygote Count Ashkenazi Jewish  \\\n",
              " 0    ...                           3470                                 0   \n",
              " 1    ...                           3470                                 0   \n",
              " 3    ...                           3472                                 0   \n",
              " 29   ...                           3472                                 0   \n",
              " 30   ...                           3472                                 0   \n",
              " 31   ...                           3470                                 0   \n",
              " 33   ...                           3472                                 0   \n",
              " 34   ...                           3472                                 0   \n",
              " 35   ...                           3472                                 0   \n",
              " 36   ...                           3472                                 0   \n",
              " 37   ...                           3470                                 0   \n",
              " 54   ...                           3468                                 0   \n",
              " 55   ...                           3470                                 0   \n",
              " 56   ...                           3472                                 0   \n",
              " 85   ...                           3472                                 0   \n",
              " 102  ...                           3472                                 0   \n",
              " 116  ...                           3470                                 0   \n",
              " 117  ...                           3472                                 0   \n",
              " 118  ...                           3470                                 0   \n",
              " 119  ...                           3472                                 0   \n",
              " 120  ...                           3468                                 0   \n",
              " 124  ...                           3468                                 0   \n",
              " 127  ...                           3472                                 0   \n",
              " 128  ...                           3472                                 0   \n",
              " 138  ...                           3470                                 0   \n",
              " 139  ...                           3470                                 0   \n",
              " 140  ...                           3466                                 0   \n",
              " 142  ...                           3472                                 0   \n",
              " 143  ...                           3466                                 0   \n",
              " 144  ...                           3470                                 0   \n",
              " 145  ...                           3470                                 0   \n",
              " 215  ...                           3468                                 0   \n",
              " 221  ...                           3470                                 0   \n",
              " 228  ...                           3470                                 0   \n",
              " \n",
              "     Hemizygote Count Ashkenazi Jewish Allele Count European (non-Finnish)  \\\n",
              " 0                                   0                                   1   \n",
              " 1                                   0                                   0   \n",
              " 3                                   0                                   1   \n",
              " 29                                  0                                   2   \n",
              " 30                                  0                                   1   \n",
              " 31                                  0                                   1   \n",
              " 33                                  0                                   1   \n",
              " 34                                  0                                   1   \n",
              " 35                                  0                                   3   \n",
              " 36                                  0                                   1   \n",
              " 37                                  0                                   0   \n",
              " 54                                  0                                   1   \n",
              " 55                                  0                                   0   \n",
              " 56                                  0                                  79   \n",
              " 85                                  0                                   1   \n",
              " 102                                 0                                   1   \n",
              " 116                                 0                                   1   \n",
              " 117                                 0                                   2   \n",
              " 118                                 0                                   1   \n",
              " 119                                 0                                   0   \n",
              " 120                                 0                                   1   \n",
              " 124                                 0                                   1   \n",
              " 127                                 0                                   1   \n",
              " 128                                 0                                   0   \n",
              " 138                                 0                                   1   \n",
              " 139                                 0                                   1   \n",
              " 140                                 0                                   0   \n",
              " 142                                 0                                   1   \n",
              " 143                                 0                                   1   \n",
              " 144                                 0                                   0   \n",
              " 145                                 0                                   2   \n",
              " 215                                 0                                   1   \n",
              " 221                                 0                                   9   \n",
              " 228                                 0                                   0   \n",
              " \n",
              "      Allele Number European (non-Finnish)  \\\n",
              " 0                                   68006   \n",
              " 1                                   68028   \n",
              " 3                                   68032   \n",
              " 29                                  68030   \n",
              " 30                                  68026   \n",
              " 31                                  68024   \n",
              " 33                                  68012   \n",
              " 34                                  68020   \n",
              " 35                                  68032   \n",
              " 36                                  68020   \n",
              " 37                                  68010   \n",
              " 54                                  68032   \n",
              " 55                                  68028   \n",
              " 56                                  68026   \n",
              " 85                                  68046   \n",
              " 102                                 68040   \n",
              " 116                                 68006   \n",
              " 117                                 68010   \n",
              " 118                                 68008   \n",
              " 119                                 68006   \n",
              " 120                                 68014   \n",
              " 124                                 68020   \n",
              " 127                                 68028   \n",
              " 128                                 68036   \n",
              " 138                                 68012   \n",
              " 139                                 67980   \n",
              " 140                                 67960   \n",
              " 142                                 68012   \n",
              " 143                                 67922   \n",
              " 144                                 68040   \n",
              " 145                                 68040   \n",
              " 215                                 68036   \n",
              " 221                                 68032   \n",
              " 228                                 68030   \n",
              " \n",
              "      Homozygote Count European (non-Finnish)  \\\n",
              " 0                                          0   \n",
              " 1                                          0   \n",
              " 3                                          0   \n",
              " 29                                         0   \n",
              " 30                                         0   \n",
              " 31                                         0   \n",
              " 33                                         0   \n",
              " 34                                         0   \n",
              " 35                                         0   \n",
              " 36                                         0   \n",
              " 37                                         0   \n",
              " 54                                         0   \n",
              " 55                                         0   \n",
              " 56                                         0   \n",
              " 85                                         0   \n",
              " 102                                        0   \n",
              " 116                                        0   \n",
              " 117                                        0   \n",
              " 118                                        0   \n",
              " 119                                        0   \n",
              " 120                                        0   \n",
              " 124                                        0   \n",
              " 127                                        0   \n",
              " 128                                        0   \n",
              " 138                                        0   \n",
              " 139                                        0   \n",
              " 140                                        0   \n",
              " 142                                        0   \n",
              " 143                                        0   \n",
              " 144                                        0   \n",
              " 145                                        0   \n",
              " 215                                        0   \n",
              " 221                                        0   \n",
              " 228                                        0   \n",
              " \n",
              "      Hemizygote Count European (non-Finnish)  PTM  \\\n",
              " 0                                          0    s   \n",
              " 1                                          0    s   \n",
              " 3                                          0    s   \n",
              " 29                                         0    s   \n",
              " 30                                         0    s   \n",
              " 31                                         0    s   \n",
              " 33                                         0    s   \n",
              " 34                                         0    s   \n",
              " 35                                         0    s   \n",
              " 36                                         0    s   \n",
              " 37                                         0    s   \n",
              " 54                                         0    s   \n",
              " 55                                         0    s   \n",
              " 56                                         0    s   \n",
              " 85                                         0    s   \n",
              " 102                                        0    s   \n",
              " 116                                        0    s   \n",
              " 117                                        0    s   \n",
              " 118                                        0    s   \n",
              " 119                                        0    s   \n",
              " 120                                        0    s   \n",
              " 124                                        0    s   \n",
              " 127                                        0    s   \n",
              " 128                                        0    s   \n",
              " 138                                        0    s   \n",
              " 139                                        0    s   \n",
              " 140                                        0    s   \n",
              " 142                                        0    s   \n",
              " 143                                        0    s   \n",
              " 144                                        0    s   \n",
              " 145                                        0    s   \n",
              " 215                                        0    s   \n",
              " 221                                        0    s   \n",
              " 228                                        0    s   \n",
              " \n",
              "                                                PTM_INF  \\\n",
              " 0    [[[PHOSPHORYLATION, PSP, NM_000484, 761, 0]], ...   \n",
              " 1        [[[PHOSPHORYLATION, PSP, NM_000484, 761, 1]]]   \n",
              " 3    [[[UBIQUITINATION, PSP, NM_000484, 751, 1]], [...   \n",
              " 29   [[[SUMOYLATION, PSP, NM_000484, 670, 1]], [[SU...   \n",
              " 30   [[[O-GalNAc, PSP, NM_000484, 667, 1], [O-GLYCO...   \n",
              " 31   [[[O-GalNAc, PSP, NM_000484, 667, 1], [O-GLYCO...   \n",
              " 33   [[[O-GalNAc, PSP, NM_000484, 663, -1], [O-GLYC...   \n",
              " 34   [[[O-GalNAc, PSP, NM_000484, 663, 1], [O-GLYCO...   \n",
              " 35   [[[O-GalNAc, PSP, NM_000484, 659, -1], [O-GLYC...   \n",
              " 36   [[[O-GalNAc, PSP, NM_000484, 659, 1], [O-GLYCO...   \n",
              " 37   [[[O-GalNAc, PSP, NM_000484, 656, -1], [O-GLYC...   \n",
              " 54             [[[O-GalNAc, PSP, NM_000484, 600, -1]]]   \n",
              " 55             [[[O-GalNAc, PSP, NM_000484, 600, -1]]]   \n",
              " 56              [[[O-GalNAc, PSP, NM_000484, 600, 1]]]   \n",
              " 85      [[[N-GLYCOSYLATION, unip, NM_000484, 542, 1]]]   \n",
              " 102     [[[PHOSPHORYLATION, unip, NM_000484, 497, 1]]]   \n",
              " 116    [[[MONO-METHYLATION, PSP, NM_000484, 468, -1]]]   \n",
              " 117    [[[MONO-METHYLATION, PSP, NM_000484, 468, -1]]]   \n",
              " 118     [[[MONO-METHYLATION, PSP, NM_000484, 468, 0]]]   \n",
              " 119     [[[MONO-METHYLATION, PSP, NM_000484, 468, 0]]]   \n",
              " 120     [[[MONO-METHYLATION, PSP, NM_000484, 468, 1]]]   \n",
              " 124  [[[PHOSPHORYLATION, PSP, NM_000484, 441, -1], ...   \n",
              " 127       [[[UBIQUITINATION, PSP, NM_000484, 429, 0]]]   \n",
              " 128       [[[UBIQUITINATION, PSP, NM_000484, 429, 0]]]   \n",
              " 138  [[[PHOSPHORYLATION, PSP, NM_000484, 381, -1], ...   \n",
              " 139  [[[PHOSPHORYLATION, PSP, NM_000484, 381, 0], [...   \n",
              " 140  [[[PHOSPHORYLATION, PSP, NM_000484, 381, 1], [...   \n",
              " 142  [[[O-GalNAc, PSP, NM_000484, 370, 0]], [[PHOSP...   \n",
              " 143  [[[O-GalNAc, PSP, NM_000484, 366, 0]], [[O-Gal...   \n",
              " 144      [[[UBIQUITINATION, PSP, NM_000484, 363, -1]]]   \n",
              " 145      [[[UBIQUITINATION, PSP, NM_000484, 363, -1]]]   \n",
              " 215  [[[PHOSPHORYLATION, PSP, NM_000484, 206, 0], [...   \n",
              " 221  [[[PHOSPHORYLATION, PSP, NM_000484, 198, 0], [...   \n",
              " 228  [[[ACETYLATION, PSP, NM_000484, 178, 0], [UBIQ...   \n",
              " \n",
              "                                              Variantes  \n",
              " 0    [[NM_000484.4, 761], [NM_201414.3, 686], [NM_2...  \n",
              " 1    [[NM_000484.4, 760], [NM_201414.3, 685], [NM_2...  \n",
              " 3    [[NM_000484.4, 750], [NM_201414.3, 675], [NM_2...  \n",
              " 29   [[NM_000484.4, 669], [NM_201414.3, 594], [NM_2...  \n",
              " 30   [[NM_000484.4, 666], [NM_201414.3, 591], [NM_2...  \n",
              " 31   [[NM_000484.4, 666], [NM_201414.3, 591], [NM_2...  \n",
              " 33   [[NM_000484.4, 664], [NM_201414.3, 589], [NM_2...  \n",
              " 34   [[NM_000484.4, 662], [NM_201414.3, 587], [NM_2...  \n",
              " 35   [[NM_000484.4, 660], [NM_201414.3, 585], [NM_2...  \n",
              " 36   [[NM_000484.4, 658], [NM_201414.3, 583], [NM_2...  \n",
              " 37   [[NM_000484.4, 657], [NM_201414.3, 582], [NM_2...  \n",
              " 54   [[NM_000484.4, 601], [NM_201414.3, 526], [NM_2...  \n",
              " 55   [[NM_000484.4, 601], [NM_201414.3, 526], [NM_2...  \n",
              " 56   [[NM_000484.4, 599], [NM_201414.3, 524], [NM_2...  \n",
              " 85   [[NM_000484.4, 541], [NM_201414.3, 466], [NM_2...  \n",
              " 102  [[NM_000484.4, 496], [NM_201414.3, 421], [NM_2...  \n",
              " 116  [[NM_000484.4, 469], [NM_201414.3, 394], [NM_2...  \n",
              " 117  [[NM_000484.4, 469], [NM_201414.3, 394], [NM_2...  \n",
              " 118  [[NM_000484.4, 468], [NM_201414.3, 393], [NM_2...  \n",
              " 119  [[NM_000484.4, 468], [NM_201414.3, 393], [NM_2...  \n",
              " 120  [[NM_000484.4, 467], [NM_201414.3, 392], [NM_2...  \n",
              " 124  [[NM_000484.4, 442], [NM_201414.3, 367], [NM_2...  \n",
              " 127  [[NM_000484.4, 429], [NM_201414.3, 354], [NM_2...  \n",
              " 128  [[NM_000484.4, 429], [NM_201414.3, 354], [NM_2...  \n",
              " 138  [[NM_000484.4, 382], [NM_201414.3, 307], [NM_2...  \n",
              " 139  [[NM_000484.4, 381], [NM_201414.3, 306], [NM_2...  \n",
              " 140  [[NM_000484.4, 380], [NM_201414.3, 305], [NM_2...  \n",
              " 142  [[NM_000484.4, 370], [NM_201414.3, 295], [NM_2...  \n",
              " 143  [[NM_000484.4, 366], [NM_201414.3, 291], [NM_2...  \n",
              " 144  [[NM_000484.4, 364], [NM_001136130.3, 308], [N...  \n",
              " 145  [[NM_000484.4, 364], [NM_001136130.3, 308], [N...  \n",
              " 215  [[NM_000484.4, 206], [NM_201414.3, 206], [NM_2...  \n",
              " 221  [[NM_000484.4, 198], [NM_201414.3, 198], [NM_2...  \n",
              " 228  [[NM_000484.4, 178], [NM_201414.3, 178], [NM_2...  \n",
              " \n",
              " [34 rows x 64 columns],\n",
              " 'MAPT':      Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
              " 0            17  45962347   rs974837695         C         A  gnomAD Genomes   \n",
              " 1            17  45962348  rs1477365646         C         T  gnomAD Genomes   \n",
              " 6            17  45962387   rs144611688         C         T  gnomAD Genomes   \n",
              " 7            17  45962392   rs746904464         G         A  gnomAD Genomes   \n",
              " 8            17  45962393  rs1568240688         G         A  gnomAD Genomes   \n",
              " ..          ...       ...           ...       ...       ...             ...   \n",
              " 351          17  46024088   rs768841567         G         A  gnomAD Genomes   \n",
              " 352          17  46024097   rs748310006         G         A  gnomAD Genomes   \n",
              " 354          17  46024115    rs63750191         C         A  gnomAD Genomes   \n",
              " 356          17  46024125    rs63750991         C         T  gnomAD Genomes   \n",
              " 357          17  46024152           NaN         T         A  gnomAD Genomes   \n",
              " \n",
              "      Filters - exomes Filters - genomes          Transcript HGVS Consequence  \\\n",
              " 0                 NaN              PASS  ENST00000262410.10        p.Pro4Thr   \n",
              " 1                 NaN              PASS  ENST00000262410.10        p.Pro4Leu   \n",
              " 6                 NaN              PASS  ENST00000262410.10       p.Thr17Met   \n",
              " 7                 NaN              PASS  ENST00000262410.10       p.Gly19Arg   \n",
              " 8                 NaN              PASS  ENST00000262410.10       p.Gly19Glu   \n",
              " ..                ...               ...                 ...              ...   \n",
              " 351               NaN              PASS  ENST00000262410.10      p.Gly807Ser   \n",
              " 352               NaN              PASS  ENST00000262410.10      p.Asp810Asn   \n",
              " 354               NaN              PASS  ENST00000262410.10      p.Gln816Lys   \n",
              " 356               NaN              PASS  ENST00000262410.10      p.Thr819Met   \n",
              " 357               NaN              PASS  ENST00000262410.10      p.Leu828Gln   \n",
              " \n",
              "      ... Allele Number Ashkenazi Jewish Homozygote Count Ashkenazi Jewish  \\\n",
              " 0    ...                           3472                                 0   \n",
              " 1    ...                           3472                                 0   \n",
              " 6    ...                           3468                                 0   \n",
              " 7    ...                           3470                                 0   \n",
              " 8    ...                           3472                                 0   \n",
              " ..   ...                            ...                               ...   \n",
              " 351  ...                           3470                                 0   \n",
              " 352  ...                           3468                                 0   \n",
              " 354  ...                           3470                                 0   \n",
              " 356  ...                           3470                                 0   \n",
              " 357  ...                           3470                                 0   \n",
              " \n",
              "     Hemizygote Count Ashkenazi Jewish Allele Count European (non-Finnish)  \\\n",
              " 0                                   0                                   0   \n",
              " 1                                   0                                   1   \n",
              " 6                                   0                                   1   \n",
              " 7                                   0                                   1   \n",
              " 8                                   0                                   0   \n",
              " ..                                ...                                 ...   \n",
              " 351                                 0                                   2   \n",
              " 352                                 0                                   1   \n",
              " 354                                 0                                   3   \n",
              " 356                                 0                                   2   \n",
              " 357                                 0                                   1   \n",
              " \n",
              "      Allele Number European (non-Finnish)  \\\n",
              " 0                                   68006   \n",
              " 1                                   68002   \n",
              " 6                                   68010   \n",
              " 7                                   68016   \n",
              " 8                                   68038   \n",
              " ..                                    ...   \n",
              " 351                                 68030   \n",
              " 352                                 68038   \n",
              " 354                                 68038   \n",
              " 356                                 68030   \n",
              " 357                                 68022   \n",
              " \n",
              "      Homozygote Count European (non-Finnish)  \\\n",
              " 0                                          0   \n",
              " 1                                          0   \n",
              " 6                                          0   \n",
              " 7                                          0   \n",
              " 8                                          0   \n",
              " ..                                       ...   \n",
              " 351                                        0   \n",
              " 352                                        0   \n",
              " 354                                        0   \n",
              " 356                                        0   \n",
              " 357                                        0   \n",
              " \n",
              "      Hemizygote Count European (non-Finnish)  PTM  \\\n",
              " 0                                          0    s   \n",
              " 1                                          0    s   \n",
              " 6                                          0    s   \n",
              " 7                                          0    s   \n",
              " 8                                          0    s   \n",
              " ..                                       ...  ...   \n",
              " 351                                        0    s   \n",
              " 352                                        0    s   \n",
              " 354                                        0    s   \n",
              " 356                                        0    s   \n",
              " 357                                        0    s   \n",
              " \n",
              "                                                PTM_INF  \\\n",
              " 0            [[[ACETYLATION, unip, NM_016835, 2, -1]]]   \n",
              " 1            [[[ACETYLATION, unip, NM_016835, 2, -1]]]   \n",
              " 6         [[[PHOSPHORYLATION, PSP, NM_005910, 17, 1]]]   \n",
              " 7    [[[PHOSPHORYLATION, PSP, NM_005910, 17, -1]], ...   \n",
              " 8    [[[PHOSPHORYLATION, PSP, NM_005910, 17, -1]], ...   \n",
              " ..                                                 ...   \n",
              " 351  [[[PHOSPHORYLATION, PSP, NM_005910, 413, -1], ...   \n",
              " 352  [[[PHOSPHORYLATION, PSP, NM_005910, 416, -1], ...   \n",
              " 354  [[[PHOSPHORYLATION, PSP, NM_005910, 422, -1], ...   \n",
              " 356  [[[PHOSPHORYLATION, PSP, NM_005910, 427, 1], [...   \n",
              " 357      [[[PHOSPHORYLATION, PSP, NM_005910, 435, 0]]]   \n",
              " \n",
              "                                              Variantes  \n",
              " 0    [[NM_005910.6, 3], [NM_016834.5, 3], [NM_01683...  \n",
              " 1    [[NM_005910.6, 3], [NM_016834.5, 3], [NM_01683...  \n",
              " 6    [[NM_005910.6, 16], [NM_016834.5, 16], [NM_016...  \n",
              " 7    [[NM_005910.6, 18], [NM_016834.5, 18], [NM_016...  \n",
              " 8    [[NM_005910.6, 18], [NM_016834.5, 18], [NM_016...  \n",
              " ..                                                 ...  \n",
              " 351  [[NM_005910.6, 414], [NM_016834.5, 356], [NM_0...  \n",
              " 352  [[NM_005910.6, 417], [NM_016834.5, 359], [NM_0...  \n",
              " 354  [[NM_005910.6, 423], [NM_016834.5, 365], [NM_0...  \n",
              " 356  [[NM_005910.6, 426], [NM_016834.5, 368], [NM_0...  \n",
              " 357  [[NM_005910.6, 435], [NM_016834.5, 377], [NM_0...  \n",
              " \n",
              " [99 rows x 64 columns],\n",
              " 'PSEN1':      Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
              " 25           14  73170836  rs1198845230         A         C  gnomAD Genomes   \n",
              " 29           14  73170860           NaN         T         G  gnomAD Genomes   \n",
              " 30           14  73170861  rs1475323603         C         G  gnomAD Genomes   \n",
              " 37           14  73170939           NaN         A         G  gnomAD Genomes   \n",
              " 99           14  73206449   rs115865530         A         G  gnomAD Genomes   \n",
              " 100          14  73206470    rs17125721         A         G  gnomAD Genomes   \n",
              " 107          14  73211847  rs1018763711         A         G  gnomAD Genomes   \n",
              " 108          14  73211854   rs767146336         T         A  gnomAD Genomes   \n",
              " 111          14  73211867   rs571825723         C         T  gnomAD Genomes   \n",
              " 112          14  73211868   rs149299753         G         A  gnomAD Genomes   \n",
              " 113          14  73211874    rs63751164         C         T  gnomAD Genomes   \n",
              " 114          14  73211876   rs376433615         C         G  gnomAD Genomes   \n",
              " 115          14  73211876   rs376433615         C         T  gnomAD Genomes   \n",
              " 116          14  73211886    rs63751174         G         A  gnomAD Genomes   \n",
              " 117          14  73211886    rs63751174         G         C  gnomAD Genomes   \n",
              " 121          14  73211906   rs200888596         T         G  gnomAD Genomes   \n",
              " 122          14  73211912  rs1485025978         A         G  gnomAD Genomes   \n",
              " 123          14  73211913           NaN         G         T  gnomAD Genomes   \n",
              " \n",
              "      Filters - exomes Filters - genomes          Transcript HGVS Consequence  \\\n",
              " 25                NaN              PASS  ENST00000324501.10       p.Ser43Arg   \n",
              " 29                NaN              PASS  ENST00000324501.10       p.Ser51Ala   \n",
              " 30                NaN              PASS  ENST00000324501.10       p.Ser51Cys   \n",
              " 37                NaN              PASS  ENST00000324501.10       p.Tyr77Cys   \n",
              " 99                NaN              PASS  ENST00000324501.10      p.Lys311Arg   \n",
              " 100               NaN              PASS  ENST00000324501.10      p.Glu318Gly   \n",
              " 107               NaN              PASS  ENST00000324501.10      p.Asp345Gly   \n",
              " 108               NaN              PASS  ENST00000324501.10      p.His347Gln   \n",
              " 111               NaN              PASS  ENST00000324501.10      p.Arg352Cys   \n",
              " 112               NaN              PASS  ENST00000324501.10      p.Arg352His   \n",
              " 113               NaN              PASS  ENST00000324501.10      p.Thr354Ile   \n",
              " 114               NaN              PASS  ENST00000324501.10      p.Pro355Ala   \n",
              " 115               NaN              PASS  ENST00000324501.10      p.Pro355Ser   \n",
              " 116               NaN              PASS  ENST00000324501.10      p.Arg358Gln   \n",
              " 117               NaN              PASS  ENST00000324501.10      p.Arg358Pro   \n",
              " 121               NaN              PASS  ENST00000324501.10      p.Ser365Ala   \n",
              " 122               NaN              PASS  ENST00000324501.10      p.Ser367Gly   \n",
              " 123               NaN              PASS  ENST00000324501.10      p.Ser367Ile   \n",
              " \n",
              "      ... Allele Number Ashkenazi Jewish Homozygote Count Ashkenazi Jewish  \\\n",
              " 25   ...                           3470                                 0   \n",
              " 29   ...                           3472                                 0   \n",
              " 30   ...                           3472                                 0   \n",
              " 37   ...                           3472                                 0   \n",
              " 99   ...                           3470                                 0   \n",
              " 100  ...                           3470                                 0   \n",
              " 107  ...                           3456                                 0   \n",
              " 108  ...                           3460                                 0   \n",
              " 111  ...                           3450                                 0   \n",
              " 112  ...                           3464                                 0   \n",
              " 113  ...                           3468                                 0   \n",
              " 114  ...                           3466                                 0   \n",
              " 115  ...                           3466                                 0   \n",
              " 116  ...                           3466                                 0   \n",
              " 117  ...                           3466                                 0   \n",
              " 121  ...                           3464                                 0   \n",
              " 122  ...                           3466                                 0   \n",
              " 123  ...                           3466                                 0   \n",
              " \n",
              "     Hemizygote Count Ashkenazi Jewish Allele Count European (non-Finnish)  \\\n",
              " 25                                  0                                   2   \n",
              " 29                                  0                                   1   \n",
              " 30                                  0                                   1   \n",
              " 37                                  0                                   1   \n",
              " 99                                  0                                   2   \n",
              " 100                                 0                                1434   \n",
              " 107                                 0                                   2   \n",
              " 108                                 0                                   0   \n",
              " 111                                 0                                   0   \n",
              " 112                                 0                                   1   \n",
              " 113                                 0                                   0   \n",
              " 114                                 0                                   0   \n",
              " 115                                 0                                   1   \n",
              " 116                                 0                                   4   \n",
              " 117                                 0                                   1   \n",
              " 121                                 0                                   2   \n",
              " 122                                 0                                   2   \n",
              " 123                                 0                                   0   \n",
              " \n",
              "      Allele Number European (non-Finnish)  \\\n",
              " 25                                  68026   \n",
              " 29                                  68024   \n",
              " 30                                  68022   \n",
              " 37                                  68044   \n",
              " 99                                  68048   \n",
              " 100                                 68038   \n",
              " 107                                 67786   \n",
              " 108                                 67788   \n",
              " 111                                 67444   \n",
              " 112                                 68008   \n",
              " 113                                 68020   \n",
              " 114                                 68000   \n",
              " 115                                 68000   \n",
              " 116                                 67996   \n",
              " 117                                 67996   \n",
              " 121                                 67998   \n",
              " 122                                 68004   \n",
              " 123                                 67984   \n",
              " \n",
              "      Homozygote Count European (non-Finnish)  \\\n",
              " 25                                         0   \n",
              " 29                                         0   \n",
              " 30                                         0   \n",
              " 37                                         0   \n",
              " 99                                         0   \n",
              " 100                                       15   \n",
              " 107                                        0   \n",
              " 108                                        0   \n",
              " 111                                        0   \n",
              " 112                                        0   \n",
              " 113                                        0   \n",
              " 114                                        0   \n",
              " 115                                        0   \n",
              " 116                                        0   \n",
              " 117                                        0   \n",
              " 121                                        0   \n",
              " 122                                        0   \n",
              " 123                                        0   \n",
              " \n",
              "      Hemizygote Count European (non-Finnish)  PTM  \\\n",
              " 25                                         0    s   \n",
              " 29                                         0    s   \n",
              " 30                                         0    s   \n",
              " 37                                         0    s   \n",
              " 99                                         0    s   \n",
              " 100                                        0    s   \n",
              " 107                                        0    s   \n",
              " 108                                        0    s   \n",
              " 111                                        0    s   \n",
              " 112                                        0    s   \n",
              " 113                                        0    s   \n",
              " 114                                        0    s   \n",
              " 115                                        0    s   \n",
              " 116                                        0    s   \n",
              " 117                                        0    s   \n",
              " 121                                        0    s   \n",
              " 122                                        0    s   \n",
              " 123                                        0    s   \n",
              " \n",
              "                                                PTM_INF  \\\n",
              " 25   [[[PHOSPHORYLATION, PSP, NM_000021, 43, 1], [P...   \n",
              " 29   [[[PHOSPHORYLATION, PSP, NM_000021, 51, 1], [P...   \n",
              " 30   [[[PHOSPHORYLATION, PSP, NM_000021, 51, 1], [P...   \n",
              " 37         [[[UBIQUITINATION, PSP, NM_000021, 76, 0]]]   \n",
              " 99   [[[PHOSPHORYLATION, PSP, NM_000021, 310, -1], ...   \n",
              " 100  [[[PHOSPHORYLATION, PSP, NM_000021, 319, 1], [...   \n",
              " 107  [[[PHOSPHORYLATION, PSP, NM_000021, 346, 1], [...   \n",
              " 108  [[[PHOSPHORYLATION, PSP, NM_000021, 346, -1], ...   \n",
              " 111  [[[PHOSPHORYLATION, PSP, NM_000021, 353, 1], [...   \n",
              " 112  [[[PHOSPHORYLATION, PSP, NM_000021, 353, 1], [...   \n",
              " 113  [[[PHOSPHORYLATION, PSP, NM_000021, 353, -1], ...   \n",
              " 114  [[[PHOSPHORYLATION, PSP, NM_000021, 354, -1], ...   \n",
              " 115  [[[PHOSPHORYLATION, PSP, NM_000021, 354, -1], ...   \n",
              " 116  [[[PHOSPHORYLATION, PSP, NM_000021, 357, -1], ...   \n",
              " 117  [[[PHOSPHORYLATION, PSP, NM_000021, 357, -1], ...   \n",
              " 121  [[[PHOSPHORYLATION, PSP, NM_000021, 365, 0], [...   \n",
              " 122  [[[PHOSPHORYLATION, PSP, NM_000021, 366, -1], ...   \n",
              " 123  [[[PHOSPHORYLATION, PSP, NM_000021, 366, -1], ...   \n",
              " \n",
              "                                     Variantes  \n",
              " 25     [[NM_000021.4, 42], [NM_007318.3, 38]]  \n",
              " 29     [[NM_000021.4, 50], [NM_007318.3, 46]]  \n",
              " 30     [[NM_000021.4, 50], [NM_007318.3, 46]]  \n",
              " 37     [[NM_000021.4, 76], [NM_007318.3, 72]]  \n",
              " 99   [[NM_000021.4, 311], [NM_007318.3, 307]]  \n",
              " 100  [[NM_000021.4, 318], [NM_007318.3, 314]]  \n",
              " 107  [[NM_000021.4, 345], [NM_007318.3, 341]]  \n",
              " 108  [[NM_000021.4, 347], [NM_007318.3, 343]]  \n",
              " 111  [[NM_000021.4, 352], [NM_007318.3, 348]]  \n",
              " 112  [[NM_000021.4, 352], [NM_007318.3, 348]]  \n",
              " 113  [[NM_000021.4, 354], [NM_007318.3, 350]]  \n",
              " 114  [[NM_000021.4, 355], [NM_007318.3, 351]]  \n",
              " 115  [[NM_000021.4, 355], [NM_007318.3, 351]]  \n",
              " 116  [[NM_000021.4, 358], [NM_007318.3, 354]]  \n",
              " 117  [[NM_000021.4, 358], [NM_007318.3, 354]]  \n",
              " 121  [[NM_000021.4, 365], [NM_007318.3, 361]]  \n",
              " 122  [[NM_000021.4, 367], [NM_007318.3, 363]]  \n",
              " 123  [[NM_000021.4, 367], [NM_007318.3, 363]]  \n",
              " \n",
              " [18 rows x 64 columns],\n",
              " 'PSEN2':      Chromosome   Position         rsIDs Reference Alternate          Source  \\\n",
              " 17            1  226881933   rs200878942         G         A  gnomAD Genomes   \n",
              " 23            1  226881972   rs552406611         C         T  gnomAD Genomes   \n",
              " 24            1  226881981           NaN         G         A  gnomAD Genomes   \n",
              " 25            1  226881984   rs781116134         C         T  gnomAD Genomes   \n",
              " 26            1  226881986   rs147326133         A         G  gnomAD Genomes   \n",
              " 27            1  226881987           NaN         C         A  gnomAD Genomes   \n",
              " 28            1  226881987   rs149354305         C         T  gnomAD Genomes   \n",
              " 29            1  226881990   rs748504448         C         T  gnomAD Genomes   \n",
              " 30            1  226881992   rs142892469         C         T  gnomAD Genomes   \n",
              " 31            1  226881993   rs771375988         G         A  gnomAD Genomes   \n",
              " 39            1  226883709  rs1051303953         G         A  gnomAD Genomes   \n",
              " 40            1  226883712   rs143501870         A         G  gnomAD Genomes   \n",
              " 69            1  226883907           NaN         A         C  gnomAD Genomes   \n",
              " 137           1  226890124   rs199689738         A         T  gnomAD Genomes   \n",
              " 142           1  226891305   rs200501733         C         T  gnomAD Genomes   \n",
              " 143           1  226891305   rs200501733         C         A  gnomAD Genomes   \n",
              " 152           1  226891754   rs994711686         T         C  gnomAD Genomes   \n",
              " 153           1  226891772   rs777639021         C         A  gnomAD Genomes   \n",
              " 154           1  226891772   rs777639021         C         G  gnomAD Genomes   \n",
              " 155           1  226891773    rs63750207         C         G  gnomAD Genomes   \n",
              " \n",
              "      Filters - exomes Filters - genomes         Transcript HGVS Consequence  \\\n",
              " 17                NaN              PASS  ENST00000366783.8        p.Ser9Asn   \n",
              " 23                NaN              PASS  ENST00000366783.8       p.Ser22Leu   \n",
              " 24                NaN              PASS  ENST00000366783.8       p.Ser25Asn   \n",
              " 25                NaN              PASS  ENST00000366783.8       p.Pro26Leu   \n",
              " 26                NaN              PASS  ENST00000366783.8       p.Thr27Ala   \n",
              " 27                NaN              PASS  ENST00000366783.8       p.Thr27Lys   \n",
              " 28                NaN              PASS  ENST00000366783.8       p.Thr27Met   \n",
              " 29                NaN              PASS  ENST00000366783.8       p.Pro28Leu   \n",
              " 30                NaN              PASS  ENST00000366783.8       p.Arg29Cys   \n",
              " 31                NaN              PASS  ENST00000366783.8       p.Arg29His   \n",
              " 39                NaN              PASS  ENST00000366783.8       p.Ser49Asn   \n",
              " 40                NaN              PASS  ENST00000366783.8       p.Gln50Arg   \n",
              " 69                NaN              PASS  ENST00000366783.8      p.Lys115Thr   \n",
              " 137               NaN              PASS  ENST00000366783.8      p.Ile293Leu   \n",
              " 142               NaN              PASS  ENST00000366783.8      p.Ala305Val   \n",
              " 143               NaN              PASS  ENST00000366783.8      p.Ala305Glu   \n",
              " 152               NaN              PASS  ENST00000366783.8      p.Tyr328His   \n",
              " 153               NaN              PASS  ENST00000366783.8      p.Pro334Thr   \n",
              " 154               NaN              PASS  ENST00000366783.8      p.Pro334Ala   \n",
              " 155               NaN              PASS  ENST00000366783.8      p.Pro334Arg   \n",
              " \n",
              "      ... Allele Number Ashkenazi Jewish Homozygote Count Ashkenazi Jewish  \\\n",
              " 17   ...                           3470                                 0   \n",
              " 23   ...                           3472                                 0   \n",
              " 24   ...                           3472                                 0   \n",
              " 25   ...                           3472                                 0   \n",
              " 26   ...                           3470                                 0   \n",
              " 27   ...                           3472                                 0   \n",
              " 28   ...                           3472                                 0   \n",
              " 29   ...                           3472                                 0   \n",
              " 30   ...                           3472                                 0   \n",
              " 31   ...                           3470                                 0   \n",
              " 39   ...                           3472                                 0   \n",
              " 40   ...                           3472                                 0   \n",
              " 69   ...                           3362                                 0   \n",
              " 137  ...                           3468                                 0   \n",
              " 142  ...                           3470                                 0   \n",
              " 143  ...                           3470                                 0   \n",
              " 152  ...                           3468                                 0   \n",
              " 153  ...                           3472                                 0   \n",
              " 154  ...                           3472                                 0   \n",
              " 155  ...                           3472                                 0   \n",
              " \n",
              "     Hemizygote Count Ashkenazi Jewish Allele Count European (non-Finnish)  \\\n",
              " 17                                  0                                   1   \n",
              " 23                                  0                                   2   \n",
              " 24                                  0                                   0   \n",
              " 25                                  0                                   0   \n",
              " 26                                  0                                   2   \n",
              " 27                                  0                                   0   \n",
              " 28                                  0                                   4   \n",
              " 29                                  0                                   1   \n",
              " 30                                  0                                   0   \n",
              " 31                                  0                                   0   \n",
              " 39                                  0                                   1   \n",
              " 40                                  0                                   6   \n",
              " 69                                  0                                   1   \n",
              " 137                                 0                                   1   \n",
              " 142                                 0                                   0   \n",
              " 143                                 0                                   0   \n",
              " 152                                 0                                   1   \n",
              " 153                                 0                                   0   \n",
              " 154                                 0                                   0   \n",
              " 155                                 0                                   1   \n",
              " \n",
              "      Allele Number European (non-Finnish)  \\\n",
              " 17                                  68038   \n",
              " 23                                  68036   \n",
              " 24                                  68038   \n",
              " 25                                  68032   \n",
              " 26                                  68028   \n",
              " 27                                  68040   \n",
              " 28                                  68040   \n",
              " 29                                  68038   \n",
              " 30                                  68032   \n",
              " 31                                  68036   \n",
              " 39                                  68042   \n",
              " 40                                  68040   \n",
              " 69                                  65222   \n",
              " 137                                 68028   \n",
              " 142                                 68000   \n",
              " 143                                 68000   \n",
              " 152                                 68020   \n",
              " 153                                 68020   \n",
              " 154                                 68020   \n",
              " 155                                 68028   \n",
              " \n",
              "      Homozygote Count European (non-Finnish)  \\\n",
              " 17                                         0   \n",
              " 23                                         0   \n",
              " 24                                         0   \n",
              " 25                                         0   \n",
              " 26                                         0   \n",
              " 27                                         0   \n",
              " 28                                         0   \n",
              " 29                                         0   \n",
              " 30                                         0   \n",
              " 31                                         0   \n",
              " 39                                         0   \n",
              " 40                                         0   \n",
              " 69                                         0   \n",
              " 137                                        0   \n",
              " 142                                        0   \n",
              " 143                                        0   \n",
              " 152                                        0   \n",
              " 153                                        0   \n",
              " 154                                        0   \n",
              " 155                                        0   \n",
              " \n",
              "      Hemizygote Count European (non-Finnish)  PTM  \\\n",
              " 17                                         0    s   \n",
              " 23                                         0    s   \n",
              " 24                                         0    s   \n",
              " 25                                         0    s   \n",
              " 26                                         0    s   \n",
              " 27                                         0    s   \n",
              " 28                                         0    s   \n",
              " 29                                         0    s   \n",
              " 30                                         0    s   \n",
              " 31                                         0    s   \n",
              " 39                                         0    s   \n",
              " 40                                         0    s   \n",
              " 69                                         0    s   \n",
              " 137                                        0    s   \n",
              " 142                                        0    s   \n",
              " 143                                        0    s   \n",
              " 152                                        0    s   \n",
              " 153                                        0    s   \n",
              " 154                                        0    s   \n",
              " 155                                        0    s   \n",
              " \n",
              "                                                PTM_INF  \\\n",
              " 17   [[[PHOSPHORYLATION, PSP, NM_000447, 7, -1], [P...   \n",
              " 23   [[[PHOSPHORYLATION, PSP, NM_000447, 22, 1], [P...   \n",
              " 24   [[[PHOSPHORYLATION, PSP, NM_000447, 25, 1], [P...   \n",
              " 25   [[[PHOSPHORYLATION, PSP, NM_000447, 25, 0], [P...   \n",
              " 26   [[[PHOSPHORYLATION, PSP, NM_000447, 25, -1], [...   \n",
              " 27   [[[PHOSPHORYLATION, PSP, NM_000447, 25, -1], [...   \n",
              " 28   [[[PHOSPHORYLATION, PSP, NM_000447, 25, -1], [...   \n",
              " 29   [[[PHOSPHORYLATION, PSP, NM_000447, 27, 0], [P...   \n",
              " 30   [[[PHOSPHORYLATION, PSP, NM_000447, 27, -1], [...   \n",
              " 31   [[[PHOSPHORYLATION, PSP, NM_000447, 27, -1], [...   \n",
              " 39        [[[PHOSPHORYLATION, PSP, NM_000447, 49, 1]]]   \n",
              " 40        [[[PHOSPHORYLATION, PSP, NM_000447, 49, 0]]]   \n",
              " 69        [[[UBIQUITINATION, PSP, NM_000447, 115, 1]]]   \n",
              " 137      [[[PHOSPHORYLATION, PSP, NM_000447, 294, 1]]]   \n",
              " 142          [[[ACETYLATION, PSP, NM_000447, 306, 1]]]   \n",
              " 143          [[[ACETYLATION, PSP, NM_000447, 306, 1]]]   \n",
              " 152  [[[PHOSPHORYLATION, PSP, NM_000447, 327, -1], ...   \n",
              " 153  [[[PHOSPHORYLATION, PSP, NM_000447, 335, 1], [...   \n",
              " 154  [[[PHOSPHORYLATION, PSP, NM_000447, 335, 1], [...   \n",
              " 155  [[[PHOSPHORYLATION, PSP, NM_000447, 335, 1], [...   \n",
              " \n",
              "                                     Variantes  \n",
              " 17       [[NM_000447.3, 8], [NM_012486.3, 8]]  \n",
              " 23     [[NM_000447.3, 21], [NM_012486.3, 21]]  \n",
              " 24     [[NM_000447.3, 24], [NM_012486.3, 24]]  \n",
              " 25     [[NM_000447.3, 25], [NM_012486.3, 25]]  \n",
              " 26     [[NM_000447.3, 26], [NM_012486.3, 26]]  \n",
              " 27     [[NM_000447.3, 26], [NM_012486.3, 26]]  \n",
              " 28     [[NM_000447.3, 26], [NM_012486.3, 26]]  \n",
              " 29     [[NM_000447.3, 27], [NM_012486.3, 27]]  \n",
              " 30     [[NM_000447.3, 28], [NM_012486.3, 28]]  \n",
              " 31     [[NM_000447.3, 28], [NM_012486.3, 28]]  \n",
              " 39     [[NM_000447.3, 48], [NM_012486.3, 48]]  \n",
              " 40     [[NM_000447.3, 49], [NM_012486.3, 49]]  \n",
              " 69   [[NM_000447.3, 114], [NM_012486.3, 114]]  \n",
              " 137  [[NM_000447.3, 293], [NM_012486.3, 293]]  \n",
              " 142  [[NM_000447.3, 305], [NM_012486.3, 305]]  \n",
              " 143  [[NM_000447.3, 305], [NM_012486.3, 305]]  \n",
              " 152  [[NM_000447.3, 328], [NM_012486.3, 327]]  \n",
              " 153  [[NM_000447.3, 334], [NM_012486.3, 333]]  \n",
              " 154  [[NM_000447.3, 334], [NM_012486.3, 333]]  \n",
              " 155  [[NM_000447.3, 334], [NM_012486.3, 333]]  \n",
              " \n",
              " [20 rows x 64 columns],\n",
              " 'TREM2':      Chromosome  Position         rsIDs Reference Alternate          Source  \\\n",
              " 83            6  41161385  rs1271841898         C         T  gnomAD Genomes   \n",
              " 84            6  41161391  rs1483052893         G         A  gnomAD Genomes   \n",
              " 85            6  41161395   rs142232675         C         T  gnomAD Genomes   \n",
              " 88            6  41161416   rs774054773         C         T  gnomAD Genomes   \n",
              " 115           6  41161595   rs534417680         T         G  gnomAD Genomes   \n",
              " 116           6  41161599   rs764738032         G         A  gnomAD Genomes   \n",
              " \n",
              "      Filters - exomes Filters - genomes         Transcript HGVS Consequence  \\\n",
              " 83                NaN              PASS  ENST00000373113.8       p.Gly90Asp   \n",
              " 84                NaN              PASS  ENST00000373113.8       p.Thr88Ile   \n",
              " 85                NaN              PASS  ENST00000373113.8       p.Asp87Asn   \n",
              " 88                NaN              PASS  ENST00000373113.8       p.Gly80Arg   \n",
              " 115               NaN              PASS  ENST00000373113.8       p.Asn20Thr   \n",
              " 116               NaN              PASS  ENST00000373113.8       p.His19Tyr   \n",
              " \n",
              "      ... Allele Number Ashkenazi Jewish Homozygote Count Ashkenazi Jewish  \\\n",
              " 83   ...                           3472                                 0   \n",
              " 84   ...                           3472                                 0   \n",
              " 85   ...                           3472                                 0   \n",
              " 88   ...                           3470                                 0   \n",
              " 115  ...                           3470                                 0   \n",
              " 116  ...                           3468                                 0   \n",
              " \n",
              "     Hemizygote Count Ashkenazi Jewish Allele Count European (non-Finnish)  \\\n",
              " 83                                  0                                   1   \n",
              " 84                                  0                                   0   \n",
              " 85                                  0                                 193   \n",
              " 88                                  0                                   0   \n",
              " 115                                 0                                   1   \n",
              " 116                                 0                                   2   \n",
              " \n",
              "      Allele Number European (non-Finnish)  \\\n",
              " 83                                  68040   \n",
              " 84                                  68046   \n",
              " 85                                  68042   \n",
              " 88                                  68046   \n",
              " 115                                 68002   \n",
              " 116                                 68030   \n",
              " \n",
              "      Homozygote Count European (non-Finnish)  \\\n",
              " 83                                         0   \n",
              " 84                                         0   \n",
              " 85                                         3   \n",
              " 88                                         0   \n",
              " 115                                        0   \n",
              " 116                                        0   \n",
              " \n",
              "      Hemizygote Count European (non-Finnish)  PTM  \\\n",
              " 83                                         0    s   \n",
              " 84                                         0    s   \n",
              " 85                                         0    s   \n",
              " 88                                         0    s   \n",
              " 115                                        0    s   \n",
              " 116                                        0    s   \n",
              " \n",
              "                                             PTM_INF  \\\n",
              " 83     [[[PHOSPHORYLATION, PSP, NM_018965, 92, 1]]]   \n",
              " 84    [[[PHOSPHORYLATION, PSP, NM_018965, 88, -1]]]   \n",
              " 85     [[[PHOSPHORYLATION, PSP, NM_018965, 88, 1]]]   \n",
              " 88   [[[N-GLYCOSYLATION, unip, NM_018965, 79, -1]]]   \n",
              " 115  [[[N-GLYCOSYLATION, unip, NM_018965, 20, -1]]]   \n",
              " 116   [[[N-GLYCOSYLATION, unip, NM_018965, 20, 1]]]   \n",
              " \n",
              "                                      Variantes  \n",
              " 83   [[NM_018965.4, 91], [NM_001271821.2, 91]]  \n",
              " 84   [[NM_018965.4, 89], [NM_001271821.2, 89]]  \n",
              " 85   [[NM_018965.4, 87], [NM_001271821.2, 87]]  \n",
              " 88   [[NM_018965.4, 80], [NM_001271821.2, 80]]  \n",
              " 115  [[NM_018965.4, 21], [NM_001271821.2, 21]]  \n",
              " 116  [[NM_018965.4, 19], [NM_001271821.2, 19]]  \n",
              " \n",
              " [6 rows x 64 columns]}"
            ]
          },
          "metadata": {},
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import files\n",
        "df_resul_gnomad.to_csv('df_resul_gnomad.csv') \n",
        "files.download('df_resul_gnomad.csv')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "YjGYwjEasvGA",
        "outputId": "ac95b7d8-b4b1-4f2e-d3bf-c1d28113c975"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_fc598158-1397-486c-b758-7d7d96a4b7cb\", \"df_resul_gnomad.csv\", 193)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "variantes=pd.read_csv(\"/content/drive/MyDrive/Projeto/Projeto/sabe2.txt\", sep='\\t')\n",
        "dic_re1={}\n",
        "df_resul_Sabe=pd.DataFrame(columns=[\"Gene\",\"Sabe Total\",\"Sabe_ptm\",\"Sabe_Porc\"])\n",
        "\n",
        "for chave in dic :\n",
        "  gene=chave\n",
        "  variantes1=variantes[variantes[\"Gene\"]==gene]\n",
        "  final=tab_final(gene)\n",
        "  final\n",
        "  list_x=variantes1[\"posi\"].tolist()\n",
        "  \n",
        "  tk=[]\n",
        "  tk1=[]\n",
        "  tk2=[]\n",
        "  for hgh in range(len(list_x)):\n",
        "    print(hgh)  \n",
        "    x=list_x[hgh]\n",
        "\n",
        "    teste1=tab_teste1(final,x)\n",
        "    teste1\n",
        "    df=tab_exons_seq(teste1)\n",
        "    lista=resul(teste1,df,x)\n",
        "\n",
        "    a,b=sitios(lista,gene)\n",
        "    tk+=[a]\n",
        "    tk1+=[b]\n",
        "    tk2+=[lista]\n",
        "  print(tk)\n",
        "  variantes1[\"PTM\"]=tk\n",
        "  variantes1[\"PTM_INF\"]=tk1\n",
        "  variantes1[\"Variantes\"]=tk2\n",
        "  variantes_sito=variantes1[variantes1[\"PTM\"]==\"s\"]\n",
        "  dic_re1[chave]=variantes_sito\n",
        "  X=len(variantes1)\n",
        "  X1=len(variantes_sito)\n",
        "  X2=(X1*100)/X\n",
        "  X,X1,X2\n",
        "  lg=[chave,X,X1,X2]\n",
        "  df_resul_Sabe.loc[len(df_resul_Sabe)] = lg\n",
        "  "
      ],
      "metadata": {
        "id": "NnTzuIYTZCwz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df_resul_Sabe"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 206
        },
        "id": "iBpf-H7nkHlf",
        "outputId": "385d9ceb-2a93-4e63-d91e-5424bddc1120"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    Gene Sabe Total Sabe_ptm  Sabe_Porc\n",
              "0    APP         11        1   9.090909\n",
              "1   MAPT         31        6  19.354839\n",
              "2  PSEN1          4        0   0.000000\n",
              "3  PSEN2         11        0   0.000000\n",
              "4  TREM2         16        1   6.250000"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-6980cee3-9a3f-46ef-b262-ce2696793739\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Gene</th>\n",
              "      <th>Sabe Total</th>\n",
              "      <th>Sabe_ptm</th>\n",
              "      <th>Sabe_Porc</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>APP</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>9.090909</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>MAPT</td>\n",
              "      <td>31</td>\n",
              "      <td>6</td>\n",
              "      <td>19.354839</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PSEN1</td>\n",
              "      <td>4</td>\n",
              "      <td>0</td>\n",
              "      <td>0.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PSEN2</td>\n",
              "      <td>11</td>\n",
              "      <td>0</td>\n",
              "      <td>0.000000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>TREM2</td>\n",
              "      <td>16</td>\n",
              "      <td>1</td>\n",
              "      <td>6.250000</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-6980cee3-9a3f-46ef-b262-ce2696793739')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-6980cee3-9a3f-46ef-b262-ce2696793739 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-6980cee3-9a3f-46ef-b262-ce2696793739');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 18
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import files\n",
        "df_resul_Sabe.to_csv('df_resul_Sabe.csv') \n",
        "files.download('df_resul_Sabe.csv')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "9Qmz05OgtCmx",
        "outputId": "fe6e406b-9c5e-4369-8a5a-4b0b9596b05d"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_5724d03c-8c1f-47bc-aad1-bf63979887eb\", \"df_resul_Sabe.csv\", 146)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "dic_re1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XcqoqDOXp0ue",
        "outputId": "fb6fde58-500a-4097-ebea-bf7f43ccacf3"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'APP':       Chro      posi Ref Al            Conse_P Gene  \\\n",
              " 19398   21  25911855   C  T  nonsynonymous SNV  APP   \n",
              " \n",
              "                                               Transcrip.     avsnp147 FILTER  \\\n",
              " 19398  APP:NM_001136129:exon11:c.G1402A:p.E468K,APP:N...  rs140304729   PASS   \n",
              " \n",
              "       CEGH_Filter  HomozygousALT_count  Hemizygous_count  Allele_number  \\\n",
              " 19398         vSR                    0               0.0           2342   \n",
              " \n",
              "        Allele_ALT_count  Frequencies         Cohort PTM  \\\n",
              " 19398               2.0     0.000854  SABE-WGS-1171   s   \n",
              " \n",
              "                                       PTM_INF  \\\n",
              " 19398  [[[O-GalNAc, PSP, NM_000484, 600, 1]]]   \n",
              " \n",
              "                                                Variantes  \n",
              " 19398  [[NM_000484.4, 599], [NM_201414.3, 524], [NM_2...  ,\n",
              " 'MAPT':        Chro      posi Ref Al            Conse_P  Gene  \\\n",
              " 160412   17  45983441   G  A  nonsynonymous SNV  MAPT   \n",
              " 160415   17  45983912   C  T  nonsynonymous SNV  MAPT   \n",
              " 160426   17  45971888   G  A  nonsynonymous SNV  MAPT   \n",
              " 160431   17  45996468   G  A  nonsynonymous SNV  MAPT   \n",
              " 160435   17  45991482   T  C  nonsynonymous SNV  MAPT   \n",
              " 160439   17  45996557   G  A  nonsynonymous SNV  MAPT   \n",
              " \n",
              "                                                Transcrip.      avsnp147  \\\n",
              " 160412  MAPT:NM_001123066:exon6:c.G637A:p.G213R,MAPT:N...    rs76375268   \n",
              " 160415  MAPT:NM_001123066:exon6:c.C1108T:p.R370W,MAPT:...    rs17651549   \n",
              " 160426  MAPT:NM_001123066:exon3:c.G163A:p.G55R,MAPT:NM...  rs1012826460   \n",
              " 160431  MAPT:NM_016834:exon6:c.G452A:p.R151H,MAPT:NM_0...   rs115492908   \n",
              " 160435  MAPT:NM_016834:exon5:c.T278C:p.I93T,MAPT:NM_01...   rs114635790   \n",
              " 160439  MAPT:NM_016834:exon6:c.G541A:p.A181T,MAPT:NM_0...    rs63750096   \n",
              " \n",
              "        FILTER CEGH_Filter  HomozygousALT_count  Hemizygous_count  \\\n",
              " 160412   PASS          SR                    0               0.0   \n",
              " 160415   PASS         vSR                   54               0.0   \n",
              " 160426   PASS         vSR                    0               0.0   \n",
              " 160431   PASS         vSR                    0               0.0   \n",
              " 160435   PASS         vSR                    0               0.0   \n",
              " 160439   PASS         vSR                    0               0.0   \n",
              " \n",
              "         Allele_number  Allele_ALT_count  Frequencies         Cohort PTM  \\\n",
              " 160412           2342              29.0     0.012383  SABE-WGS-1171   s   \n",
              " 160415           2342             468.0     0.199829  SABE-WGS-1171   s   \n",
              " 160426           2342               1.0     0.000427  SABE-WGS-1171   s   \n",
              " 160431           2342               1.0     0.000427  SABE-WGS-1171   s   \n",
              " 160435           2342               3.0     0.001281  SABE-WGS-1171   s   \n",
              " 160439           2342               3.0     0.001281  SABE-WGS-1171   s   \n",
              " \n",
              "                                                   PTM_INF  \\\n",
              " 160412  [[[PHOSPHORYLATION, PSP, NM_016835, 214, 1], [...   \n",
              " 160415  [[[PHOSPHORYLATION, PSP, NM_016835, 369, -1], ...   \n",
              " 160426  [[[PHOSPHORYLATION, PSP, NM_005910, 56, 1]], [...   \n",
              " 160431  [[[PHOSPHORYLATION, PSP, NM_005910, 208, -1], ...   \n",
              " 160435  [[[PHOSPHORYLATION, PSP, NM_005910, 149, -1]],...   \n",
              " 160439  [[[PHOSPHORYLATION, PSP, NM_005910, 238, -1], ...   \n",
              " \n",
              "                                                 Variantes  \n",
              " 160412        [[NM_016835.5, 213], [NM_001123066.4, 213]]  \n",
              " 160415        [[NM_016835.5, 370], [NM_001123066.4, 370]]  \n",
              " 160426  [[NM_005910.6, 55], [NM_016835.5, 55], [NM_001...  \n",
              " 160431  [[NM_005910.6, 209], [NM_016834.5, 151], [NM_0...  \n",
              " 160435  [[NM_005910.6, 150], [NM_016834.5, 92], [NM_01...  \n",
              " 160439  [[NM_005910.6, 239], [NM_016834.5, 181], [NM_0...  ,\n",
              " 'PSEN1': Empty DataFrame\n",
              " Columns: [Chro, posi, Ref, Al, Conse_P, Gene, Transcrip., avsnp147, FILTER, CEGH_Filter, HomozygousALT_count, Hemizygous_count, Allele_number, Allele_ALT_count, Frequencies, Cohort, PTM, PTM_INF, Variantes]\n",
              " Index: [],\n",
              " 'PSEN2': Empty DataFrame\n",
              " Columns: [Chro, posi, Ref, Al, Conse_P, Gene, Transcrip., avsnp147, FILTER, CEGH_Filter, HomozygousALT_count, Hemizygous_count, Allele_number, Allele_ALT_count, Frequencies, Cohort, PTM, PTM_INF, Variantes]\n",
              " Index: [],\n",
              " 'TREM2':        Chro      posi Ref Al            Conse_P   Gene  \\\n",
              " 293760    6  41161395   C  T  nonsynonymous SNV  TREM2   \n",
              " \n",
              "                                                Transcrip.     avsnp147 FILTER  \\\n",
              " 293760  TREM2:NM_001271821:exon2:c.G259A:p.D87N,TREM2:...  rs142232675   PASS   \n",
              " \n",
              "        CEGH_Filter  HomozygousALT_count  Hemizygous_count  Allele_number  \\\n",
              " 293760         vSR                    0               0.0           2342   \n",
              " \n",
              "         Allele_ALT_count  Frequencies         Cohort PTM  \\\n",
              " 293760               5.0     0.002135  SABE-WGS-1171   s   \n",
              " \n",
              "                                              PTM_INF  \\\n",
              " 293760  [[[PHOSPHORYLATION, PSP, NM_018965, 88, 1]]]   \n",
              " \n",
              "                                         Variantes  \n",
              " 293760  [[NM_018965.4, 87], [NM_001271821.2, 87]]  }"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "Vqme0-tbJW4d"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        ""
      ],
      "metadata": {
        "id": "caOula-qp5dx"
      },
      "execution_count": 20,
      "outputs": []
    }
  ]
}